src/v/config/base_property.h

```diff
@@ -107,6 +107,7 @@ class base_property {
     virtual void reset() = 0;
     virtual bool is_default() const = 0;
     virtual bool is_hidden() const = 0;
+    virtual bool is_enterprise_only() const = 0;
```

Comment:
is there somewhere to read more about the plans? this feels like it might not be quite the right place to encode this information.

Reply:
eh, i wouldn't say there's much of a concrete plan. just messing around. 

> might not be quite the right place

you mean in the property hierarchy generally, or specifically this virtual function?

Reply:
yeh like as a property _property_

Reply:
but i dunno. it could be ok, it's just a snap reaction

Reply:
cool, that's helpful. i feel similarly fwiw. gonna do a little work outside-in to see how things line up.